plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtStopCommandWrapper.java

```diff
@@ -92,6 +93,13 @@ public Answer execute(final StopCommand command, final LibvirtComputingResource
             libvirtComputingResource.destroyNetworkRulesForVM(conn, vmName);
             final String result = libvirtComputingResource.stopVM(conn, vmName, command.isForceStop());
 
+            try {
+                LibvirtKvmAgentHook onStopHook = libvirtComputingResource.getStopHook();
+                onStopHook.handle(vmName);
+            } catch (Exception e) {
+                s_logger.warn("Exception occured when handling LibVirt VM onStop hook: {}", e);
+            }
```

Comment:
can you put this in a separate method?

Reply:
@DaanHoogland I'll handle all your comments tomorrow.

Reply:
ping, did you manage?

Reply:
@DaanHoogland fixed every your notes.